src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -624,7 +624,7 @@ void TemporalMemory::reset(void) {
   activeSegments_.clear();
   matchingSegments_.clear();
   segmentsValid_ = false;
-  anomaly_ = -1;
+  anomaly_ = -1.0f;
```

Comment:
sorry, some of these are my doing. 

- one nit, if we're already doing these changes, next time please use `static_cast<Real>(1)` instead of `1.0f`. It looks i bit off, but is futureproof. If we ever decide to play with the typedefs (say for Real), all the constants with `1.0f` would have to be manually changed again, the static_cast would be automatic. 
- and one question, for where precisely is this needed? This code has been in master and thus passed CI (even for Windows)